src/ripple/overlay/impl/TrafficCount.h

```diff
@@ -122,9 +183,8 @@ class TrafficCount
         return ret;
     }
 
-    protected:
-
-    std::map <category, TrafficStats> counts_;
+protected:
+    std::map<category, TrafficStats> counts_;
```

Comment:
Given that `category` is an enum with default values, could we use a `std::vector/std::array` instead and use the category as the index? Avoids a tree traversal/(potentially) fragmented memory. 

i.e. add a `count` member at the end of the enum class (after `unknown`), and use that to determine the size of the array/vector.

Reply:
I like that, but I'm not sure I want to do that for this PR. Let me see how it would work.